src/grelu/interpret/motifs.py

```diff
@@ -183,9 +183,9 @@ def scan_sequences(
     sites = pd.DataFrame()
     for all_idx, curr_sites in enumerate(all_sites):
         curr_sites["seq_idx"] = all_idx
```

Comment:
I think this should be `curr_sites["seq_idx"] = curr_sites.sequence_name.apply(seq_ids.index)` - so `seq_idx` should be the index of the sequence that matches the returned name.